src/Core/src/Handlers/Image/ImageHandler.Windows.cs

```diff
@@ -8,33 +9,152 @@ namespace Microsoft.Maui.Handlers
 {
 	public partial class ImageHandler : ViewHandler<IImage, Image>
 	{
+		/// <inheritdoc/>
 		protected override Image CreatePlatformView() => new Image();
 
+		/// <inheritdoc/>
 		protected override void DisconnectHandler(Image platformView)
 		{
 			base.DisconnectHandler(platformView);
 			SourceLoader.Reset();
 		}
 
+		/// <inheritdoc/>
 		public override bool NeedsContainer =>
 			VirtualView?.Background != null ||
+			VirtualView?.Aspect == Aspect.AspectFill ||
 			base.NeedsContainer;
 
+		/// <inheritdoc/>
+		protected override void SetupContainer()
+		{
+			base.SetupContainer();
+
+			// VerticalAlignment only works when the child's Height is Auto
+			PlatformView.Height = double.NaN;
+
+			MapHeight(this, VirtualView);
+			MapWidth(this, VirtualView);
+			MapMaximumHeight(this, VirtualView);
+			MapMaximumWidth(this, VirtualView);
+		}
+
+		/// <inheritdoc/>
+		protected override void RemoveContainer()
+		{
+			base.RemoveContainer();
+
+			MapHeight(this, VirtualView);
+			MapWidth(this, VirtualView);
+			MapMaximumHeight(this, VirtualView);
+			MapMaximumWidth(this, VirtualView);
+		}
+
+		/// <summary>
+		/// Maps the abstract <see cref="IView.Height"/> property to the platform-specific implementations.
+		/// </summary>
+		/// <param name="handler">The associated handler.</param>
+		/// <param name="view">The associated <see cref="Image"/> instance.</param>
+		public static void MapHeight(IImageHandler handler, IImage view)
+		{
+			// VerticalAlignment only works when the container's Height is set and the child's Height is Auto. The child's Height
+			// is set to Auto when the container is introduced
+			if (handler.ContainerView is FrameworkElement element)
+			{
+				element.Height = view.Height;
+			}
+
+			ViewHandler.MapHeight(handler, view);
```

Comment:
Won't this automatically set the child's height too? The code for MapHeight is:

```cs
((PlatformView?)handler.PlatformView)?.UpdateHeight(view);
```

If the height needs to be auto, then should this code still run?

Reply:
@hartez Should I use this pattern for all the Map* functions? I recall you saying that `			ViewHandler.MapMaximumHeight(handler, view);` should always be called regardless of the container.

Reply:
I'm wondering if you need to do anything at all with MaximumHeight and MaximumWidth. The behavior you describe in https://github.com/dotnet/maui/pull/15122#issuecomment-1679936570 is what WPF does when you specify a MaxHeight/MaxWidth on an image with UniformFill; it's probably fine that it doesn't center in that case. I think you can remove all all the max height/width mapping stuff entirely.